xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-ui/src/main/resources/XWiki/Like/Code/LiveTableResultPage.xml

```diff
@@ -0,0 +1,135 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.4" reference="XWiki.Like.Code.LiveTableResultPage" locale="">
+  <web>XWiki.Like.Code</web>
+  <name>LiveTableResultPage</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>XWiki.Like.Code</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title/>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+#if ($xcontext.action == 'get')
+  #template('hierarchy_macros.vm')
+  #if("$!{request.xpage}" == 'plain')
+    $response.setContentType('application/json')
+  #end
+  #if($request.user)
+    #set ($userReference = $services.model.resolveDocument($request.user))
+  #else
+    #set ($userReference = $services.model.resolveDocument($xcontext.userReference)) 
```

Comment:
``$xcontext.userReference`` is already a ``DocumentReference``.

Reply:
Also note that `getUserLikes` takes a `UserReference` as input. Using a document reference is a bit the old way. 

To get the current user:

`UserReference userReference = $services.user.currentUserReference`

To convert from a user as a string to a `UserReference`, well you don't need to do anything since we have a converter that converts from String to `UserReference` so you can just pass `$request.user` :)

See https://extensions.xwiki.org/xwiki/bin/view/Extension/User%20Module/API/ for more details.